File: SauceDemo/PageObjects/CheckoutPage1.cs
using OpenQA.Selenium;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SauceDemo.PageObjects
{
    public class CheckoutPage1 : BasePage
    {

        public CheckoutPage1(IWebDriver driver) : base(driver)
        {
        }

        // I'm doing it this way to show a quick way to get all the inputs as webelements.
        IList<IWebElement> inputElements => HelperMethods.FindElementsWithWait(_driver,By.ClassName("form_input"),System.TimeSpan.FromSeconds(10));
        private IWebElement BUT_continue => HelperMethods.FindElementWithWait_Clickable(_driver, By.XPath("//*[@data-test='continue']"), System.TimeSpan.FromSeconds(10));

        //I'm then going to populate the inputs from a list of data in the test
        public void EnterInfo(string f, string l, string zip)
        {
            List<string> inputValues = new List<string> { f, l, zip };
            for (int i = 0; i < Math.Min(inputElements.Count, inputValues.Count); i++)
                inputElements[i].SendKeys(inputValues[i]);
            BUT_continue.Click();
        }

        public void gotoCheckoutPage1()
        {
            Goto("checkout-step-one.html");
        }

    }
}

File: Selenium_Python/pythonProject/pages/checkoutpage1.py
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from selenium.common.exceptions import NoSuchElementException

from helper import helper
from pages.basepage import basepage

class checkoutpage1(basepage):
    def __init__(self, driver):
        super().__init__(driver)
        self.input_elements = helper.find_elements_with_wait(self._driver, By.CLASS_NAME, "form_input", 20)
        self.but_continue = helper.find_element_with_wait_clickable(self._driver, By.XPATH, "//*[@data-test='continue']", 20)

    def enter_info(self, f, l, zip_code):
        input_values = [f, l, zip_code]
        for i in range(min(len(self.input_elements), len(input_values))):
            self.input_elements[i].send_keys(input_values[i])
        self.but_continue.click()

    def goto_checkout_page1(self):
        self.goto("checkout-step-one.html")
